src/main/java/net/snowflake/ingest/streaming/internal/ClientBufferParameters.java

```diff
@@ -65,14 +73,22 @@ public ClientBufferParameters(SnowflakeStreamingIngestClientInternal clientInter
         clientInternal != null
             ? clientInternal.getParameterProvider().isEnableNewJsonParsingLogic()
             : ParameterProvider.ENABLE_NEW_JSON_PARSING_LOGIC_DEFAULT;
-    this.isIcebergMode =
-        clientInternal != null
-            ? clientInternal.isIcebergMode()
-            : ParameterProvider.IS_ICEBERG_MODE_DEFAULT;
     this.maxRowGroups =
         isIcebergMode
```

Comment:
you're reading isIcebergMode before writing to it.

Reply:
Fixed.